api/v1alpha1/sandbox_types.go

```diff
@@ -35,6 +35,18 @@ type SandboxSpec struct {
 
 // SandboxStatus defines the observed state of Sandbox.
 type SandboxStatus struct {
+	// ObservedGeneration - record generation of last successful run
```

Comment:
Nit: I _think_ it just the metadata.generation that corresponds to the status (particularly status.conditions), even if that is a failure

Reply:
removed and using status.conditions[].observedGeneration